lib/api/objectCopy.js

```diff
@@ -351,6 +356,31 @@ function objectCopy(authInfo, request, sourceBucket,
                     serverSideEncryption, destBucketMD);
             }
 
+            // using copyObject only if copying object from one external
+            // backend to the same external backend
+            if (storeMetadataParams.locationMatch && locationType &&
```

Comment:
I don't think you need to check that locationType exists here because if it doesn't the next condition will just be false

Reply:
ok thanks :)